src/input/MeshAttributes.cpp

```diff
@@ -38,250 +35,123 @@ void MeshAttributes::readXmlFile(const char* xmlFilename) {
-           lFaceIdMeshSizePropagation.push_back(faceid);
-       }
+    retrieve_attribute_and_update_class_member(&doc, "MeshSizePropagation", "ScalingFactor", &MeshSizePropagationScalingFactor);
+    retrieve_attribute_and_update_class_member(&doc, "MeshSizePropagation", "Distance", &MeshSizePropagationDistance);
+    if (auto child = doc.FirstChildElement("MeshSizePropagation")) {
+       lFaceIdMeshSizePropagation = parse_string_and_fill_list(child-> GetText());
     }
 }
 
 
 void MeshAttributes::set_regionMSize() {
-    std::string line,sval;
-    XMLElement* child;
-    XMLElement* pRoot;
-    child = doc.FirstChildElement("regionMSize");
-    for (child; child; child = child->NextSiblingElement("regionMSize"))
+    for (auto child = doc.FirstChildElement("regionMSize"); child; child = child->NextSiblingElement("regionMSize"))
     {
-       sval = child->Attribute("value");
-       double regionMSize =  atof(sval.c_str());
-       lregionMSize.push_back(regionMSize);
-       std::vector<std::string> tokens;
-       line =  child-> GetText();
-       split(tokens, line, ',');
-       std::list<int> lregionId;
-       for(int i = 0; i < tokens.size(); i++) {
-           int regionid = std::atoi(tokens[i].c_str());
-           lregionId.push_back(regionid);
-       }
-       llregionMSizeRegionId.push_back(lregionId);
+       lregionMSize.push_back( std::atof(child->Attribute("value")));
+       llregionMSizeRegionId.push_back(parse_string_and_fill_list(child-> GetText()));
     }
 }
 
 
-void MeshAttributes::set_gradation() {
-    std::string sval;
-    XMLElement* pRoot;
-    pRoot = doc.FirstChildElement("gradation");
-    if(pRoot) {
-       sval = pRoot->Attribute("value");
-       gradation =  atof(sval.c_str());
-    }
-}
+void MeshAttributes::set_global_mesh_attributes() {
+    retrieve_attribute_and_update_class_member(&doc, "gradation", "value", &gradation);
+    retrieve_attribute_and_update_class_member(&doc, "globalMSize", "value", &globalMSize);
+    retrieve_attribute_and_update_class_member(&doc, "area_AspectRatio", "value", &area_AspectRatio);
+    retrieve_attribute_and_update_class_member(&doc, "vol_AspectRatio", "value", &vol_AspectRatio);
 
-void MeshAttributes::set_globalMSize() {
-    std::string sval;
-    XMLElement* pRoot;
-    pRoot = doc.FirstChildElement("globalMSize");
-    if(pRoot) {
-       sval = pRoot->Attribute("value");
-       globalMSize =  atof(sval.c_str());
-    }
 }
 
 
-void MeshAttributes::set_area_AspectRatio() {
-    std::string sval;
-    XMLElement* pRoot;
-    pRoot = doc.FirstChildElement("area_AspectRatio");
-    if(pRoot) {
-       sval = pRoot->Attribute("value");
-       area_AspectRatio =  atof(sval.c_str());
+void MeshAttributes::retrieve_attribute_and_update_class_member(XMLNode* element, const char* sElementName, const char* sAttributeName, double* MeshAttributesMember) {
+    if (auto child = element->FirstChildElement(sElementName)) {
```

Comment:
What do we do, if element has no child with name `sElementName`? Is there some kind of error detection?

Reply:
In that case, the condition is false, and the parameter does not get updated?
Do you suggest adding a else logError(*) ?

Reply:
IIRC it is valid to not give all elements, but if an element is present, the attribute has to be there to make sense right?
So having no `gradation` tag is valid, but having a `<gradation>` tag without a `value` attribute is invalid, right?

Reply:
ha yes exactly. That what I wrote in a comment above. I will see if it is easy to implement that.